### PR TITLE
[exprtk] Update ExprTk

### PR DESCRIPTION
**Pull request by ArashPartow**
this PR is waiting for the following PR to be merged: https://github.com/microsoft/vcpkg/pull/29955

### CODE REVIEW

File: ports/exprtk/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "exprtk",
-  "version-date": "2022-01-01",
+  "version-date": "2023-01-01",
   "port-version": 2,
```
Comment (JonLiu1993) [inline ports/exprtk/vcpkg.json:4]:
```suggestion
```

Comment (JonLiu1993) [reply ports/exprtk/vcpkg.json:4]: @ArashPartow, Thanks for your PR, we should remove "port-version" when we update port.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
Please get [failure logs here](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/85402/artifacts?artifactName=failure%20logs%20for%20x86-windows&api-version=7.0&%24format=zip). Please ping me if you need any help.

File: ports/exprtk/vcpkg.json (hunk 1 above)
Comment (ArashPartow) [reply ports/exprtk/vcpkg.json]: Have now removed the port-version key from the vcpkg.json file.

**Comment by ArashPartow** [comment]
@JonLiu1993 

I'm not sure how to handle the following error as it seems to be related to another package and not ExprTk: 

> ##[error]PowerShell wrote one or more lines to the standard error stream.
> ##[error]REGRESSIONS:
> REGRESSION: vtk:x64-linux failed with BUILD_FAILED. If expected, add vtk:x64-linux=fail to /agent/_work/1/s/scripts/azure-pipelines/../ci.baseline.txt.
> 


Any ideas on how to proceed?

**Comment by JonLiu1993** [comment]
vtk depends on exprtk, and the update of exprtk makes the port that depends on it cannot find the "2.7" version port when looking for exprtk
````
-- Could NOT find ExprTk: (Required is at least version "2.7") (found D:/installed/x86-windows/include)
CMake Error at CMake/vtkModule.cmake:4579 (message):
  Could not find the ExprTk external dependency.
Call Stack (most recent call first):
  CMake/vtkModule.cmake:5173 (vtk_module_find_package)
  CMake/vtkModule.cmake:5044 (vtk_module_third_party_external)
  ThirdParty/exprtk/CMakeLists.txt:1 (vtk_module_third_party)
````

**Comment by 6ziv** [comment]
Do not update version files manually. Instead, run vcpkg x-add-versions

> @JonLiu1993 I'm not sure how to handle the following error as it seems to be related to another package and not ExprTk:
> 
> > ##[error]PowerShell wrote one or more lines to the standard error stream.
> > ##[error]REGRESSIONS:
> > REGRESSION: vtk:x64-linux failed with BUILD_FAILED. If expected, add vtk:x64-linux=fail to /agent/_work/1/s/scripts/azure-pipelines/../ci.baseline.txt.
> 
> https://dev.azure.com/vcpkg/public/_build/results?buildId=85406&view=logs&j=f79cfdd7-47a8-597f-8f57-dc3e21a8f2ad&t=da63cf11-1f12-503b-6d64-3b2fb713c172&l=7479
> 
> Any ideas on how to proceed?

Seems like a bug in vtk. In "CMake/FindExprTk" there is a hard-coded search for a regex starting with "static const char* ", however exprtk changed that to "static char_cptr version" in [a recent commit](https://github.com/ArashPartow/exprtk/commit/f46bffcd6966d38a09023fb37ba9335214c9b959#diff-58c1983215ef11e39375e044aaa946c840e57c7f72d3ba9e303ee1a9b7cd9865).
Maybe we can make a fix by removing that search (exprtk version number seems meaningless: it is always 2.71828.)

**Comment by 6ziv** [comment]
@ArashPartow I'd suggest patching the vtk port like this attachment.
[FindExprTk.patch](https://github.com/microsoft/vcpkg/files/10740966/FindExprTk.patch)

Maybe we can also patch ExprTk (maybe put the original line in comment, so it can still be matched by cmake regex search?), but it seem so strange...

**Comment by spyridon97** [comment]
In VTK have a third party repo of exprt https://gitlab.kitware.com/third-party/exprtk, in this repo with the master branch which basically the same thing as https://github.com/ArashPartow/exprtk (at some point when we pulled) and for/vtk branch in which we add some very small changes as you can see here so that it can be understood as an internal vtk module without conflict https://gitlab.kitware.com/third-party/exprtk/-/commits/for/vtk.

Would you like us to synchronize our master branch with your github branch again and rebase our for/vtk branch based on the updated master branch?

**Comment by ArashPartow** [comment]
@spyridon97  Agreed, I think it would be a good idea to update the VTK's ExprTk repo.
However the main problem seems to be coming from here:

https://github.com/Kitware/VTK/blob/master/CMake/FindExprTk.cmake#L23-L25


    file(STRINGS "${ExprTk_INCLUDE_DIR}/exprtk.hpp" _exprtk_version_header
      REGEX "static const char\\* version")
    string(REGEX MATCH "static const char\\* version = \"([0-9.]+)\"" _exprtk_version_match "${_exprtk_version_header}")


The whole regex approach to getting the version needs to be rethought, at the very least the regex should perhaps be in the form of:

    static .*version \= \"2\.7([0-9.]+)\"

File: ports/exprtk/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "exprtk",
-  "version-date": "2022-01-01",
-  "port-version": 2,
+  "version-date": "2023-01-01",
```
Comment (JonLiu1993) [inline ports/exprtk/vcpkg.json:3]:
```suggestion
  "version": "0.0.2",
```

File: ports/exprtk/portfile.cmake [hunk 3]
```diff
@@ -1,9 +1,9 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO ArashPartow/exprtk
-    REF 806C519C91FD08BA4FA19380DBF3F6E42DE9E2D1
-    SHA512 A323CCAF161FD0087FD8208D1B24C2A3FD422F8875E29B22AE70E6DD2F10F396F6BF1AD36D3FFDC10D32314AE8F83749974301A349BE0F27733292BCF5193640
-    HEAD_REF master
+    REF F46BFFCD6966D38A09023FB37BA9335214C9B959
```
Comment (JonLiu1993) [inline ports/exprtk/portfile.cmake:4]:
```suggestion
    REF "${VERSION}"
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/exprtk/vcpkg.json (hunk 2 above)
Comment (ArashPartow) [reply ports/exprtk/vcpkg.json:3]: @JonLiu1993  there are several downstream repos that use the version-date semantic for selection the version they require.  

I'll look into seeing what the affect of changing that to version will be and then do an update PR.

Comment (ArashPartow) [reply ports/exprtk/vcpkg.json:3]: @JonLiu1993  is there anything else left for this PR   in order for it to be merged?

**Comment by JonLiu1993** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/29955.

**Comment by ArashPartow** [comment]
@JonLiu1993 I see, ok can we please have https://github.com/microsoft/vcpkg/pull/29955 be merged

**Comment by JonLiu1993** [comment]
> @JonLiu1993 I see, ok can we please have #30058 be merged

@ArashPartow, thanks for your contribution, don't worry, if ci passes, we will merge as soon as possible after review.

**Comment by JonLiu1993** [comment]
Pinging @ArashPartow for response. Is work still being done for this PR?

**Comment by PhoebeHui** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/30058.

**Comment by JonLiu1993** [comment]
Convert this PR to draft since there is no progress. Please ping us if this PR is ready for review again.

**Comment by dan-shaw** [comment]
Closing this PR since there is no progress for a few months. Feel free to open a new PR if you are still working on it

**Comment by PhoebeHui** [comment]
Hi @ArashPartow, I can't reopened this PR, since the Repo had been deleted, please submit a new PR instead.

**Pull request closed without merging**